lib/tasks/stop_data_convert_and_import.rake

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+namespace :bus_stops do
+  task import_old_data: :environment do
+    csv = CSV.parse(File.read('old_stop_data.csv'), headers: true)
+    csv.each do |row|
+      import_data = row.to_hash
+      new_attributes = {}
+      conversion_methods = { 'has_power' =>
+                                 (import_data['has_power'] == 'true' ? 'Yes - Stub up' : 'No'),
+                             'shared_sign_post_frta' =>
+                                 (import_data['shared_sign_post'] == 'Yes - FRTA'),
+                             'system_map_exists' =>
+                                 (import_data['system_map_exists'] == 'true' ? 'Old map' : 'No map'),
+                             'trash' => (import_data['trash'] != 'None') }
+      conversion_methods.each do |attribute, conversion|
+        if attribute == 'shared_sign_post_frta'
+          new_attributes[attribute] = conversion unless import_data['shared_sign_post'].nil?
+        else
+          new_attributes[attribute] = conversion unless import_data[attribute].nil?
+        end
+      end
```

Comment:
I still think this is overengineering. IMO it would be easier to read to just write:

```ruby
new_attributes['has_power'] = case import_data['has_power']
                              when true then 'Yes - Stub up'
                              when false then 'No'
                              end
```

etc. It's a little less DRY, but there's *no* special logic at all built in. You can read it once and understand exactly what's going on.

Especially important since this is something we will only have to look back it if something else has gone wrong. Those are the moments when clever code is your worst enemy.